gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilExtension.kt

```diff
@@ -25,4 +25,13 @@ open class AnvilExtension {
    * to `true`, then also [generateDaggerFactories] must be set to `true`.
    */
   var generateDaggerFactoriesOnly = false
+
+  /**
+   * Enabling this indicates that only code generation should run and no transformations. This is
+   * useful for cases where you want to use `@ContributesTo`, `@ContributesBinding`, or similar
+   * annotations but _not_ `@MergeComponent` or `@MergeSubcomponent` functionality. This allows
+   * for anvil use in projects with kapt enabled but _not_ require disabling incremental compilation
+   * in kapt stub generation tasks.
+   */
+  var generationOnly = false
```

Comment:
How about `contributeOnly`? That would fit better with the language Anvil is using.

Reply:
maybe, but would that be confusing with generateDaggerFactories? Since both could be enabled at the same time. Maybe `disableComponentMerging` instead to be ultra explicit?

Reply:
Would sort of line up semantically with the `AnvilMergeAnnotationDetectorCheck` too

Reply:
Tried that out and thought it read quite well. Let me know what you think 0fbccbb

Reply:
Reads well to me as well and is very explicit.

> Since both could be enabled at the same time.

Really? That doesn't makes sense. If you generate Dagger factories only, then it's redundant to set `disableComponentMerging` to true. I wonder if this even should be a check. Only one of the flags should be set to true.

Reply:
Another point: if you generate dagger factories only, then the Gradle plugin doesn't even add the Anvil annotations to the compile classpath.

Reply:
Could be a follow up, what do you think?

Reply:
Ah no i just generate factories, not generate factories _only_. Does that make sense?

Reply:
Oh, damn. I somehow assumed you meant `generateDaggerFactoriesOnly`, but you clearly said `generateDaggerFactories`. Makes sense now 🤦🏻